libr/anal/meta.c

```diff
@@ -232,10 +232,14 @@ R_API int r_meta_del(RAnal *a, int type, ut64 addr, ut64 size) {
 	if (type == R_META_TYPE_ANY) {
 		/* special case */
 		r_meta_del (a, R_META_TYPE_COMMENT, addr, size);
+		r_meta_del (a, R_META_TYPE_VARTYPE, addr, size);
+	}
+	if (type == R_META_TYPE_COMMENT || type == R_META_TYPE_VARTYPE) {
+		snprintf (key, sizeof (key)-1, "meta.%c.0x%"PFMT64x, type, addr);
```

Comment:
is this needed only for VARTYPE or also for other types as well?

Reply:
This is needed only for metas set up through r_meta_set_string (which currently are only COMMENT and VARTYPE). All other metas, which are set up through r_meta_add, do not need this.
This was already done for COMMENT, I only extended for VARTYPE too. I think it was originally done because r_meta_set_string is faster than r_meta_add, but I'm not 100% sure.

Reply:
Where do you use r_meta_set_string for VARTYPE?

Reply:
Inside libr/core/anal_tp.c, in function type_match()

Reply:
I could modify and use r_meta_add there too, but originally (before my last pr) it was already r_meta_set_string with COMMENT, I only switched it to VARTYPE

Reply:
It's ok. Just wanted to understand ;)